### PR TITLE
libdeflate 

### PR DESCRIPTION
**Pull request by Jarred-Sumner**
### What does this PR do?

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [ ] Code changes

### CODE REVIEW

File: src/bun.js/rare_data.zig [hunk 1]
```diff
@@ -392,3 +392,14 @@ pub fn nodeFSStatWatcherScheduler(rare: *RareData, vm: *JSC.VirtualMachine) *Sta
         return rare.node_fs_stat_watcher_scheduler.?;
     };
 }
+
+pub fn deinit(this: *RareData) void {
```
Comment (Jarred-Sumner) [inline src/bun.js/rare_data.zig:396]: small memory leak fix for worker

File: packages/bun-types/bun.d.ts [hunk 2]
```diff
@@ -3481,6 +3481,13 @@ declare module "bun" {
      * Filtered data consists mostly of small values with a somewhat random distribution.
      */
     strategy?: number;
+
+    library?: "zlib";
```
Comment (dylan-conway) [inline packages/bun-types/bun.d.ts:3485]: is `"libdeflate"` the other option here? we should add a comment if so

Comment (Jarred-Sumner) [reply packages/bun-types/bun.d.ts:3485]: typescript autocomplete shows that

File: src/http.zig [hunk 3]
```diff
@@ -1520,40 +1540,93 @@ pub const InternalState = struct {
...
-            if (this.isDone() or error.ShortRead != err) {
-                Output.prettyErrorln("<r><red>Decompression error: {s}<r>", .{bun.asByteSlice(@errorName(err))});
-                Output.flush();
-                return err;
+        var still_needs_to_decompress = true;
+
+        if (FeatureFlags.isLibdeflateEnabled()) {
+            // Fast-path: use libdeflate
+            if (is_final_chunk and !this.is_libdeflate_fast_path_disabled and this.encoding.canUseLibDeflate() and this.isDone()) libdeflate: {
+                this.is_libdeflate_fast_path_disabled = true;
```
Comment (dylan-conway) [inline src/http.zig:1555]: why is the fast path disabled here?

File: src/http.zig [hunk 4]
```diff
@@ -3458,6 +3538,7 @@ fn handleResponseBodyChunkedEncodingFromMultiplePackets(
             this.state.received_last_chunk = true;
             _ = try this.state.processBodyBuffer(
                 buffer,
+                true,
```
Comment (dylan-conway) [inline src/http.zig:3541]: should this be
```suggestion
                this.state.encoding.isCompressed()
```

Comment (dylan-conway) [reply src/http.zig:3541]: nvm, encoding is checked in `processBodyBuffer`

File: src/http.zig (hunk 3 above)
Comment (Jarred-Sumner) [reply src/http.zig:1555]: libdeflate doesn't support streaming, so we can only use it once

**Pull request merged**